Add docs about using `dtype` (#1903)
astronomer/astro-sdk: python-sdk/docs/astro/sql/operators/load_file.rst

       :start-after: [START load_file_example_6]
       :end-before: [END load_file_example_6]

#. **load_options** - Use :ref:`load_options` to configure how the SDK loads data from your file to the dataframe.

   .. note::

>         If the Astro SDK fails to automatically load data from your file to a dataframe, configure `Link text <link URL>` in :ref:`load_options` manually specify the schema for the dataframe.

Reviewer: @jwitz was this `Link text <link URL>` intentional?

It seems our checks failed, with error:
/home/runner/work/astro-sdk/astro-sdk/python-sdk/docs/astro/sql/operators/load_file.rst:130:Unexpected indentation.

Would you like help on checking this?

>         If the Astro SDK fails to automatically load data from your file to a dataframe, configure `Link text <link URL>` in :ref:`load_options` manually specify the schema for the dataframe.

Author: ```suggestion
        If the Astro SDK fails to automatically load data from your file to a dataframe, configure `dtype <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html>` in :ref:`load_options` manually specify the schema for the dataframe.
```

>         If the Astro SDK fails to automatically load data from your file to a dataframe, configure `dtype <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html>` in :ref:`load_options` manually specify the schema for the dataframe.

Author: ```suggestion
        Depending on the file type you provide, the Astro SDK uses `read_csv <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html#pandas-read-csv>`_, `read_json <https://pandas.pydata.org/docs/reference/api/pandas.read_json.html>`_, or `read_parquet <https://pandas.pydata.org/docs/reference/api/pandas.read_parquet.html>`_ to parse your file and load it to a dataframe. If the Astro SDK fails to automatically load data from your file to a dataframe, configure `dtype` <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html> in :ref:`load_options` to manually specify the schema for the dataframe.
```

        For example, to load data from a `.csv` file with two columns, `id` and `name`, you would add the following to your code:

>         ```

Reviewer 2: I think it's worth mentioning where we use dtype, we internally pass these values to  - https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html#pandas-read-csv,  https://pandas.pydata.org/docs/reference/api/pandas.read_json.html and https://pandas.pydata.org/docs/reference/api/pandas.read_parquet.html so that people can know about there possible value and format.

Reviewer 3: +1

Reviewer 3: May be we should add couple of more examples for different file types

>         Depending on the file type you provide, the Astro SDK uses `read_csv <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html#pandas-read-csv>`_, `read_json <https://pandas.pydata.org/docs/reference/api/pandas.read_json.html>`_, or `read_parquet <https://pandas.pydata.org/docs/reference/api/pandas.read_parquet.html>`_ to parse your file and load it to a dataframe. If the Astro SDK fails to automatically load data from your file to a dataframe, configure `dtype` <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html> in :ref:`load_options` to manually specify the schema for the dataframe.

Author: ```suggestion
        Depending on the file type you provide, the Astro SDK uses `read_csv <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html#pandas-read-csv>`_, `read_json <https://pandas.pydata.org/docs/reference/api/pandas.read_json.html>`_, or `read_parquet <https://pandas.pydata.org/docs/reference/api/pandas.read_parquet.html>`_ to parse your file and load it to a dataframe. If the Astro SDK fails to automatically load data from your file to a dataframe, configure `dtype <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html>`_ in :ref:`load_options` to manually specify the schema for the dataframe.
```

       :start-after: [START load_file_example_6]
       :end-before: [END load_file_example_6]

#. **load_options** - Use :ref:`load_options` to configure how the SDK loads data from your file to the dataframe.

   .. note::

>         Depending on the file type you provide, the Astro SDK uses `read_csv <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html#pandas-read-csv>`_, `read_json <https://pandas.pydata.org/docs/reference/api/pandas.read_json.html>`_, or `read_parquet <https://pandas.pydata.org/docs/reference/api/pandas.read_parquet.html>`_ to parse your file and load it to a dataframe. If the Astro SDK fails to automatically load data from your file to a dataframe, configure `dtype <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html>`_ in :ref:`load_options` to manually specify the schema for the dataframe.

Author: ```suggestion
      Depending on the file type you provide, the Astro SDK uses `read_csv <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html#pandas-read-csv>`_, `read_json <https://pandas.pydata.org/docs/reference/api/pandas.read_json.html>`_, or `read_parquet <https://pandas.pydata.org/docs/reference/api/pandas.read_parquet.html>`_ to parse your file and load it to a dataframe. If the Astro SDK fails to automatically load data from your file to a dataframe, configure `dtype <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html>`_ in :ref:`load_options` to manually specify the schema for the dataframe.
```

>         For example, to load data from a `.csv` file with two columns, `id` and `name`, you would add the following to your code:

Author: ```suggestion
     For example, to load data from a `.csv` file with two columns, `id` and `name`, you would add the following to your code:
```

        ```
        dataframe = load_file(

>             input_file=File(path),

Reviewer: @jwitz we're super close to merging this change!
It seems the doc build is failing with an indentation error now:
```
/home/runner/work/astro-sdk/astro-sdk/python-sdk/docs/astro/sql/operators/load_file.rst:132:Unexpected indentation.
```

Let's try to get this merged next week! 🤞

Author: @tatiana Do you know anything about how the docs build? I can't tell if this is an indentation issue for rst, or if there's a style check being applied on my Python.

>       Depending on the file type you provide, the Astro SDK uses `read_csv <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html#pandas-read-csv>`_, `read_json <https://pandas.pydata.org/docs/reference/api/pandas.read_json.html>`_, or `read_parquet <https://pandas.pydata.org/docs/reference/api/pandas.read_parquet.html>`_ to parse your file and load it to a dataframe. If the Astro SDK fails to automatically load data from your file to a dataframe, configure `dtype <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html>`_ in :ref:`load_options` to manually specify the schema for the dataframe.
>      For example, to load data from a `.csv` file with two columns, `id` and `name`, you would add the following to your code:
> 
>      ```
>      dataframe = load_file(
>          input_file=File(path),
>          use_native_support=False,
>          load_options=[PandasLoadOptions(dtype={'id': int, 'name': str})],
>      )
>      ```

Reviewer: Paired with @jwitz on this:
```suggestion
      Depending on the file type you provide, the Astro SDK uses `read_csv <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html#pandas-read-csv>`_, `read_json <https://pandas.pydata.org/docs/reference/api/pandas.read_json.html>`_, or `read_parquet <https://pandas.pydata.org/docs/reference/api/pandas.read_parquet.html>`_ to parse your file and load it to a dataframe. If the Astro SDK fails to automatically load data from your file to a dataframe, configure `dtype <https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html>`_ in :ref:`load_options` to manually specify the schema for the dataframe.

      For example, to load data from a `.csv` file with two columns, `id` and `name`, you would add the following to your code:

      .. code-block:: python

         dataframe = load_file(
            input_file=File(path),
            use_native_support=False,
            load_options=[PandasLoadOptions(dtype={'id': int, 'name': str})],
         )
```
It renders like this:
<img width="720" alt="Screenshot 2023-05-12 at 11 32 01" src="https://github.com/astronomer/astro-sdk/assets/272048/16730c2b-233a-4959-8508-7f2cdedac7a3">
